File: C++/bee 1049.cpp
#include <iostream>
#include <iomanip>
#include <cmath>

using namespace std;

int main() {

    string tipo1, tipo2, tipo3, animal;
    cin >> tipo1 >> tipo2 >> tipo3;

    if (tipo1 == "vertebrado") {
        if (tipo2 == "ave") {
            if (tipo3 == "carnivoro") {
                animal = "aguia";
            } else if (tipo3 == "onivoro") {
                animal = "pomba";
            }
        } else if (tipo2 == "mamifero") {
            if (tipo3 == "onivoro") {
                animal = "homem";
            } else if (tipo3 == "herbivoro") {
                animal = "vaca";
            }
        }

    } else if (tipo1 == "invertebrado") {
        if (tipo2 == "inseto") {
            if (tipo3 == "hematofago") {
                animal = "pulga";
            } else if (tipo3 == "herbivoro") {
                animal = "lagarta";
            }
        } else if (tipo2 == "anelideo") {
            if (tipo3 == "hematofago") {
                animal = "sanguessuga";
            } else if (tipo3 == "onivoro") {
                animal = "minhoca";
            }
        }
    }

    cout << animal << endl;

    return 0;
}

File: Rust/bee_1049.rs
use std::io::{self, Read};

fn main() {
    let mut input = String::new();
    io::stdin().read_to_string(&mut input).unwrap();

    let palavras: Vec<String> = input
        .split_whitespace()
        .map(|x| x.parse().unwrap())
        .collect();

    if palavras[0] == "vertebrado" {
        if palavras[1] == "ave" {
            if palavras[2] == "carnivoro" {
                println!("aguia");
            } else {
                println!("pomba");
            }
        } else {
            if palavras[2] == "onivoro" {
                println!("homem");
            } else {
                println!("vaca");
            }
        }
    } else {
        if palavras[1] == "inseto" {
            if palavras[2] == "hematofago" {
                println!("pulga");
            } else {
                println!("lagarta");
            }
        } else {
            if palavras[2] == "hematofago" {
                println!("sanguessuga");
            } else {
                println!("minhoca");
            }
        }
    }
}
